sdk/servicebus/service-bus/src/receivers/receiver.ts

```diff
@@ -648,35 +654,19 @@ export class ServiceBusReceiverImpl implements ServiceBusReceiver {
     message: ServiceBusReceivedMessage,
     options?: DeadLetterOptions & { [key: string]: any }
   ): Promise<void> {
+    this._throwIfReceiverOrConnectionClosed();
+    throwErrorIfInvalidOperationOnMessage(message, this.receiveMode, this._context.connectionId);
     const msgImpl = message as ServiceBusMessageImpl;
     return deadLetterMessage(msgImpl, this._context, this.entityPath, options);
   }
 
   async renewMessageLock(message: ServiceBusReceivedMessage): Promise<Date> {
+    this._throwIfReceiverOrConnectionClosed();
+    throwErrorIfInvalidOperationOnMessage(message, this.receiveMode, this._context.connectionId);
+
     const msgImpl = message as ServiceBusMessageImpl;
-    if (!msgImpl.delivery) {
-      throw new Error("A peeked message does not have a lock to be renewed.");
-    }
 
     let associatedLinkName: string | undefined;
-    let error: Error | undefined;
-    if (!message.lockToken) {
-      error = new Error(
-        getErrorMessageNotSupportedInReceiveAndDeleteMode(`renew the lock on the message`)
-      );
-    } else if (msgImpl.delivery.remote_settled) {
```

Comment:
Looks like the error thrown for already settled is not retained?

Reply:
Yes, we expect the service to throw the error here

Reply:
Why not just throw if we know for sure?

Reply:
Check with the feature crew. If they agree, we can add.
cc @richardpark-msft